cloudfront/wellcomelibrary.org/edge-lambda/src/wellcomeLibraryArchiveRedirect.ts

```diff
@@ -1,15 +1,53 @@
 import { CloudFrontRequestEvent, Context } from 'aws-lambda';
-import { CloudFrontRequest } from 'aws-lambda/common/cloudfront';
+import {
+  CloudFrontRequest,
+  CloudFrontResultResponse,
+} from 'aws-lambda/common/cloudfront';
+import querystring from 'querystring';
+import { wellcomeCollectionRedirect } from './redirectHelpers';
+import { findWorkWithIdentifierValue } from './catalogueApi';
+
+async function getWorkWithId(term: string) {
+  const work = await findWorkWithIdentifierValue(term);
+  if (work) {
+    return wellcomeCollectionRedirect(`/works/${work.id}`);
+  }
+}
+
+async function redirectRequestUri(
+  request: CloudFrontRequest
+): Promise<undefined | CloudFrontResultResponse> {
+  const qs = querystring.parse(request.querystring);
+  const dsqItem = qs.dsqItem ? qs.dsqItem.toString() : undefined;
+  const dsqSearch = qs.dsqSearch ? qs.dsqSearch.toString() : undefined;
+  const dsqSearchTermsMatch = dsqSearch
+    ? dsqSearch.match(/'([^']*)'/g)
+    : undefined;
+  // The dserve app puts all test it's searching for in the format of `(Field='{term}')`
+  // so we just search for anything between `'`
+  const dsqSearchTerms = dsqSearchTermsMatch
+    ? dsqSearchTermsMatch.map((term) => term.replace(/'/g, '')).join(' ')
+    : undefined;
```

Comment:
What if you search for a string that contains a single quote? Should these only be removed from the start/end?

Reply:
This get's mangled by the DServe search, and I think that's OK
http://archives.wellcomelibrary.org/DServe/dserve.exe?dsqIni=Dserve.ini&dsqApp=Archive&dsqCmd=Show.tcl&dsqDb=Catalog&dsqPos=0&dsqSearch=%28%28text%29%3D%27-things-%27%29

Reply:
In this instance we will redirect to `query=-things-` which is fine.